android/app/src/main/java/com/futuretech/tudonum/ui/screens/SignupFragment.kt

```
@@ -0,0 +1,202 @@
+        // user taps inside the field itself.
+        editText.setOnClickListener { binding.tiDob.performClick() }
+
+        binding.tiDob.setOnClickListener {
+            val c = Calendar.getInstance()
+            DatePickerDialog(
+                requireContext(),
+                /** apply the custom theme here **/
+                R.style.DatePickerTheme,
+                { _, y, m, d ->
+                    // Format: YYYY‑MM‑DD
+                    editText.setText(String.format("%04d-%02d-%02d", y, m + 1, d))
```

detekt (1.23.8)

[warning] String.format("%04d-%02d-%02d", y, m + 1, d) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)